go/gilded-rose.go

```diff
@@ -1,57 +1,139 @@
 package main
 
+// Item represents an item for sale at the inn
 type Item struct {
 	name            string
 	sellIn, quality int
 }
 
-func UpdateQuality(items []*Item) {
-	for i := 0; i < len(items); i++ {
+// simple func to return a min of 2 integers
+// useful in this context to stop items value going over 50
+func min(x int, y int) int {
+	if x == y {
+		return x
+	}
+	if x < y {
+		return x
+	}
+	return y
+}
 
-		if items[i].name != "Aged Brie" && items[i].name != "Backstage passes to a TAFKAL80ETC concert" {
-			if items[i].quality > 0 {
-				if items[i].name != "Sulfuras, Hand of Ragnaros" {
-					items[i].quality = items[i].quality - 1
-				}
-			}
-		} else {
-			if items[i].quality < 50 {
-				items[i].quality = items[i].quality + 1
-				if items[i].name == "Backstage passes to a TAFKAL80ETC concert" {
-					if items[i].sellIn < 11 {
-						if items[i].quality < 50 {
-							items[i].quality = items[i].quality + 1
-						}
-					}
-					if items[i].sellIn < 6 {
-						if items[i].quality < 50 {
-							items[i].quality = items[i].quality + 1
-						}
-					}
-				}
-			}
-		}
+// simple func to return a max of 2 integers
+// useful in this context to stop items value going under 0
+func max(x int, y int) int {
+	if x == y {
+		return x
+	}
+	if x > y {
+		return x
+	}
+	return y
+}
 
-		if items[i].name != "Sulfuras, Hand of Ragnaros" {
-			items[i].sellIn = items[i].sellIn - 1
-		}
+// UpdateQualityBrie handles updates for aged brie
+func UpdateQualityBrie(item *Item) {
+	// always decrement sellInn
+	defer func() {
+		item.sellIn--
```

Comment:
To decrease code repetition, it could be moved to the `UpdateQuality`

Reply:
Not sure I follow you here

Reply:
Each of the quality change method has this `defer`, you could move it at the end of UpdateQuality and it would have the same effect.

Reply:
Ah, yes. I get what you are saying. Makes sense.